semeio/jobs/correlated_observations_scaling/exporter.py

```diff
@@ -0,0 +1,49 @@
+import json
+import os
+from enum import Enum
+
+import numpy as np
+
+
+class ExportKey(Enum):
+    singular_values = "singular_values"
+    nr_components = "nr_components"
+    nr_observations = "nr_observations"
+    scaled_matrix = "scaled_matrix"
+    scale_factor = "scale_factor"
+    keys = "keys"
+
+
+class NumpyEncoder(json.JSONEncoder):
+    def default(self, obj):
+        if isinstance(obj, np.ndarray):
+            return obj.tolist()
+        return json.JSONEncoder.default(self, obj)
+
+
+class Exporter(object):
+    def __init__(self, config):
+        self._data = {}
+        self._dir = config.snapshot.EXPORT_DIR
+
+    def export(self, key, data, message=None):
+        if message is not None:
```

Comment:
It seemed like this was a common thing in COS. Printing data to the terminal, and possibly having someone interpret that output into data. This combines both, for no particular reason other than it might be useful during the scaling (in e.g. the status window in the GUI), and after the workflow has ended (on disk). But I am not confident it should be a part of such an interface.

Reply:
It just seems to me like the method is trying to do two potentially unrelated things that can be left to the caller, namely printing a message and exporting some data.